### PR TITLE
Chore/fix float number yaml file

### LINKED ISSUE #1387: Chore: format model.yml and unitests
- The float number of model.yml content is not rounded
     - Need to update `0.949999988` -> `0.95`
- Code for output content of model.yml is duplicated, need to add functions to `string_utils.h` and `format_utils.h`
- Add unitests for `string_utils.h` and `format_utils.h` #1069 .

### PR DESCRIPTION
**Pull request by nguyenhoangthuan99**
Fix #1387 

Round float number when write to yaml file
image

### CODE REVIEW

File: engine/config/yaml_config.cc [hunk 1]
```diff
@@ -276,7 +276,28 @@ void YamlHandler::WriteYamlFile(const std::string& file_path) const {
...
+          std::ostringstream oss;
+          oss << std::fixed << std::setprecision(6) << doubleValue;
+          std::string strValue = oss.str();
+          // Remove trailing zeros after the decimal point
+          strValue.erase(strValue.find_last_not_of('0') + 1, std::string::npos);
+          if (strValue.back() == '.') {
+            strValue.pop_back();
+          }
+          outFile << strValue;
+        } catch (...) {
```
Comment (namchuai) [inline engine/config/yaml_config.cc]: I think `catch (const std::exception& e)` is better. Should avoid `...`

File: engine/config/model_config.h [hunk 2]
```diff
@@ -264,12 +264,19 @@ struct ModelConfig {
...
     };
 
-    // Helper function to print float values with fixed precision
+    // Updated helper function to print float values with fixed precision
     auto print_float = [&print_kv, &BLUE](const std::string& key, float value) {
       if (!std::isnan(value)) {
         std::ostringstream float_oss;
-        float_oss << std::fixed << std::setprecision(9) << value;
-        print_kv(key, float_oss.str(), BLUE);
+        float_oss << std::fixed << std::setprecision(6) << value;
```
Comment (vansangpfiev) [inline engine/config/model_config.h]: Should create a function in string_utils.h. Then we can add unit tests for it

File: engine/config/model_config.h [hunk 3]
```diff
@@ -264,12 +264,19 @@ struct ModelConfig {
       print_kv(key, value ? "true" : "false", MAGENTA);
     };
 
-    // Helper function to print float values with fixed precision
+    // Updated helper function to print float values with fixed precision
```
Comment (namchuai) [inline engine/config/model_config.h]: I think we should separate this function to `format_utils.h`

File: engine/config/yaml_config.cc [hunk 4]
```diff
@@ -291,7 +312,9 @@ void YamlHandler::WriteYamlFile(const std::string& file_path) const {
                   "Model ID which is used for request construct - should be "
                   "unique between models (author / quantization)");
     writeKeyValue("name", yaml_node_["name"], "metadata.general.name");
-    writeKeyValue("version", yaml_node_["version"], "metadata.version");
+    if(yaml_node_["version"]){
+      outFile<<"version: "<<yaml_node_["version"].as<std::string>()<<"\n";
```
Comment (namchuai) [inline engine/config/yaml_config.cc]: please format code

**Review by vansangpfiev — APPROVED** [review]

**Pull request merged**